### PR TITLE
CRM-7-Create-Entities-for-payment-service

### PR DESCRIPTION
**Pull request by Dima013**

### CODE REVIEW

File: payment-service/src/main/java/entity/Payment.java [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+public class Payment {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    @Column(name = "id", unique = true, nullable = false)
+    private Integer id;
+
+    @Column(name = "order_id", unique = true, nullable = false)
+    private UUID order_id;
+
+    @Column(name = "is_cash", length = 32)
```
Comment (polinaucc) [inline payment-service/src/main/java/entity/Payment.java:25]: why is length 32?)

File: payment-service/src/main/java/entity/Payment.java [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    private Boolean is_cash;
+
+    @Column(name = "sum", unique = true, nullable = false)
+    private Double sum;
```
Comment (polinaucc) [inline payment-service/src/main/java/entity/Payment.java:29]: it is better to take BigDecimal here. 
unique constraint is redundant

File: payment-service/src/main/java/entity/Payment.java [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+    @Column(name = "payment_date", unique = true, nullable = false)
```
Comment (polinaucc) [inline payment-service/src/main/java/entity/Payment.java:31]: why do we have unique here?

File: payment-service/src/main/java/entity/Payment.java [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+    private LocalDateTime payment_date;
+
+    @Column(name = "status", unique = true, nullable = false)
```
Comment (polinaucc) [inline payment-service/src/main/java/entity/Payment.java:34]: unique is redundant
let's make status as a enum

File: user-manager-service/pom.xml [hunk 5]
```diff
@@ -10,12 +10,13 @@
     </parent>
 
     <groupId>com.crm-food-establishment</groupId>
-    <artifactId>user-manager</artifactId>
```
Comment (polinaucc) [inline user-manager-service/pom.xml:13]: why did u make these changes to user-manager pom? =)

File: payment-service/src/main/java/com/crmfoodestablishment/paymentservice/entity/Payment.java [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+    @Column(name = "sum", unique = true, nullable = false, precision = 10, scale = 2)
+    private BigDecimal sum;
+
+    @Column(name = "payment_date", unique = true, nullable = false)
+    private LocalDateTime payment_date;
+
+    @Enumerated(EnumType.STRING)
+    @Column(name = "status", nullable = false)
+    private PaymentStage status;
+
```
Comment (polinaucc) [inline payment-service/src/main/java/com/crmfoodestablishment/paymentservice/entity/Payment.java:38]: let's remove these redundant empty lines

File: payment-service/src/main/java/com/crmfoodestablishment/paymentservice/entity/PaymentStage.java [hunk 7]
```diff
@@ -0,0 +1,17 @@
+package com.crmfoodestablishment.paymentservice.entity;
+
+public enum PaymentStage {
+    UNPAID("Unpaid"),
```
Comment (polinaucc) [inline payment-service/src/main/java/com/crmfoodestablishment/paymentservice/entity/PaymentStage.java:4]: are u sue that we need constructor here? and getDisplayNamemethod?

File: payment-service/src/main/resources/application.properties [hunk 8]
```diff
@@ -1,4 +1,12 @@
 spring.application.name=payment-service
 server.port=0
-
 eureka.client.service-url.defaultZone=http://localhost:8761/eureka/
+spring.datasource.url=jdbc:mysql://localhost:3306/libary
+spring.datasource.username=root
+spring.datasource.password=1D1234
+spring.datasource.driver-class-name=com.mysql.cj.jdbc.Driver
+spring.liquibase.change-log=classpath:db/changelog/db.changelog_master.xml
+spring.jpa.show-sql=true
```
Comment (polinaucc) [inline payment-service/src/main/resources/application.properties:9]: it seems to me that spring.jpa.show-sql as false in the main application profile

File: user-manager-service/pom.xml (hunk 5 above)
Comment (Dima013) [reply user-manager-service/pom.xml:13]: done

File: payment-service/src/main/java/com/crmfoodestablishment/paymentservice/entity/Payment.java (hunk 6 above)
Comment (Dima013) [reply payment-service/src/main/java/com/crmfoodestablishment/paymentservice/entity/Payment.java:38]: sory but is ctrl+alt+l

File: payment-service/src/main/java/com/crmfoodestablishment/paymentservice/entity/PaymentStage.java (hunk 7 above)
Comment (Dima013) [reply payment-service/src/main/java/com/crmfoodestablishment/paymentservice/entity/PaymentStage.java:4]: done

File: payment-service/src/main/resources/application.properties (hunk 8 above)
Comment (Dima013) [reply payment-service/src/main/resources/application.properties:9]: done

File: payment-service/pom.xml [hunk 9]
```diff
@@ -5,7 +5,7 @@
     <parent>
         <groupId>org.springframework.boot</groupId>
         <artifactId>spring-boot-starter-parent</artifactId>
-        <version>3.1.2</version>
+        <version>3.1.1</version>
```
Comment (momopoi443) [inline payment-service/pom.xml:8]: give arguments for 3.1.1 version or change back to 3.1.2

File: payment-service/src/main/resources/application.properties (hunk 8 above)
Comment (momopoi443) [reply payment-service/src/main/resources/application.properties:9]: spring.jpa.show-sql need to be changed to false

**Review by momopoi443 — COMMENTED** [review]
To sum up, reviewed last commit. SQL logging and version of spring boot starters need to be fixed

File: payment-service/src/main/resources/application.properties (hunk 8 above)
Comment (Dima013) [reply payment-service/src/main/resources/application.properties:9]: done

File: payment-service/pom.xml (hunk 9 above)
Comment (Dima013) [reply payment-service/pom.xml:8]: done

**Review by polinaucc — APPROVED** [review]

**Pull request merged**